packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambdaLayerWalkthrough.ts

```diff
@@ -1,122 +1,131 @@
+        _.assign(layerInputParameters, await inquirer.prompt(layerAccountAccessQuestion()));
+        break;
+      case Permissions.awsOrg:
+        _.assign(layerInputParameters, await inquirer.prompt(layerOrgAccessQuestion()));
+        break;
+    }
   }
+  _.assign(parameters, { layerVersion: '1' });
+  // add layer version to parameters
+  _.assign(parameters, { layerVersionsMap: createVersionsMap(layerInputParameters, '1') });
+  _.assign(parameters, { build: true });
   return parameters;
 }
 
-function layerNameQuestion(context: any) {
-  return [
-    {
-      type: 'input',
-      name: 'layerName',
-      message: 'Provide a name for your Lambda layer:',
-      validate: input => {
-        input = input.trim();
-        const meta = context.amplify.getProjectMeta();
-        if (!/^[a-zA-Z0-9_\-]{1,140}$/.test(input)) {
-          return 'Lambda Layer names are 1-140 characters long and can only contain letters, numbers, -, _';
-        } else if (meta.function && meta.function.hasOwnProperty(input)) {
-          return `A Lambda Layer with the name ${input} already exists in this project.`;
-        }
-        return true;
-      },
-      default: () => {
-        const appName = context.amplify
-          .getProjectDetails()
-          .projectConfig.projectName.toLowerCase()
-          .replace(/[^a-zA-Z0-9]/gi, '');
-        const [shortId] = uuid().split('-');
-        return `${appName}${shortId}`;
-      },
-    },
-  ];
-}
+export async function updateLayerWalkthrough(
+  context: any,
+  templateParameters: Partial<LayerParameters>,
+): Promise<Partial<LayerParameters>> {
+  const { allResources } = await context.amplify.getResourceStatus();
+  const resources = allResources.filter(resource => resource.service === ServiceName.LambdaLayer).map(resource => resource.resourceName);
 
-function layerPermissionsQuestion() {
-  return [
+  if (resources.length === 0) {
+    context.print.error('No Lambda Layer resource to update. Please use "amplify add function" command to create a new Function');
+    process.exit(0);
+    return;
+  }
+  const resourceQuestion = [
     {
+      name: 'resourceName',
+      message: 'Please select the Lambda Layer you want to update',
       type: 'list',
-      name: 'layerPermissions',
-      message: 'Who should have permission to use this layer?',
-      choices: [
-        {
-          name: 'Only the current AWS account',
-          value: Permissions.private,
-        },
-        {
-          name: 'Specific AWS accounts',
-          value: Permissions.awsAccounts,
-        },
-        {
-          name: 'Specific AWS organization',
-          value: Permissions.awsOrg,
-        },
-        {
-          name: 'Public (everyone on AWS can use this layer)',
-          value: Permissions.public,
-        },
-      ],
-      default: Permissions.private,
+      choices: resources,
     },
   ];
-}
+  const resourceAnswer = await inquirer.prompt(resourceQuestion);
```

Comment:
Not a huge issue, but if there's only one layer in the project, you don't need to prompt for this